Framework/API/inc/MantidAPI/AlgorithmFactoryObserver.h

```diff
@@ -0,0 +1,61 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_KERNEL_ALGORITHMFACTORYOBSERVER_H_
+#define MANTID_KERNEL_ALGORITHMFACTORYOBSERVER_H_
+
+#include "MantidAPI/AlgorithmFactory.h"
+#include "MantidAPI/DllConfig.h"
+#include "MantidKernel/DataService.h"
+#include <Poco/NObserver.h>
+
+using namespace Mantid::Kernel;
```

Comment:
Just a minor point about `using`. It's considered a little dangerous in header files as there is potential for it to introduce clashing names by just including a header file: https://stackoverflow.com/a/19940227.